src/mpid/ch4/src/ch4_types.h

```diff
@@ -102,6 +102,7 @@ typedef struct MPIDIG_put_msg_t {
     uint64_t win_id;
     MPIR_Request *preq_ptr;
     MPI_Aint target_disp;
+    MPI_Aint target_lb;
```

Comment:
This doesn't seem correct.  You already have the target flattened datatype, from which you can extract the true lb as needed.  The Typerep routines already account for lb/ub.

Reply:
this is only for the case where we don't flatten. previously, the am code would flatten any datatype that wasn't a builtin, which seems wrong to me.

Reply:
We do need to flatten any datatype that's not a builtin, do we not?

Reply:
I thought for contig types we only need the true lower bound. Flattening achieves that, yes, but it seems like overkill vs. a single `MPI_Aint` in the packet header. I suppose we can drop this if it's not that significant.

Reply:
Zero true lb does not guarantee contig.  You are talking about a case that has a non-sticky (so, not resized) and non-zero true lb, but the type is contiguous?  I'm having a hard time even constructing a datatype where that would be true.  :-)

Reply:
I am lost. Where are I using zero true lb to determine if something is contiguous?

Reply:
Oh, nevermind, I misread the code.  You are using it within the contig case.  Can you change the variable to `target_true_lb`?

Reply:
Yeah, I will change it. I'm going to shelve this change right now because it has a slight conflict with the flattening PR, and I want to prioritize the atomics changes before this.